src/core/include/openvino/op/util/symbolic_info.hpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <openvino/core/descriptor/tensor.hpp>
+
+#include "openvino/core/node.hpp"
+#include "openvino/core/runtime_attribute.hpp"
+#include "openvino/core/visibility.hpp"
+
+namespace ov {
+
+OPENVINO_API void set_up_symbolic_info(const std::shared_ptr<ov::Model>& model,
+                                       const std::shared_ptr<ov::TableOfEquivalence>& table);
+OPENVINO_API void set_up_symbolic_info(const ov::Output<ov::Node>& output,
+                                       const std::shared_ptr<ov::TableOfEquivalence>& table);
+
+OPENVINO_API void populate_tensor_with_missing_labels(ov::descriptor::Tensor& tensor);
+
+OPENVINO_API bool skip_invalidation(const ov::descriptor::Tensor& tensor);
+OPENVINO_API std::shared_ptr<ov::TableOfEquivalence> table_of_equivalence(const std::shared_ptr<ov::Model>& model);
+OPENVINO_API std::shared_ptr<ov::TableOfEquivalence> table_of_equivalence(const ov::descriptor::Tensor& tensor);
+
+OPENVINO_API void remove_symbolic_info(const std::shared_ptr<ov::Model>& model, bool outermost_model = true);
```

Comment:
It becomes a part of public API. Please provide description for each of the function. Why do we need it at the public level?

Reply:
I'd like to do it a bit later. Current PR is green -- I don't wish to rerun it just for the docs.